A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,443 @@
+management server populates fields that the client does not yet support),
+then the same scenario will cause the older clients to fail instead of
+just ignoring the new feature.
+
+Note, however, that there are some unavoidable cases where enabling a
+new feature will cause older clients to break.  These specific cases
+will be called out below.
+
+#### Initial Request on the ADS Stream
+
+The first request on the ADS stream will include a Node message that
+identifies the client.  As mentioned above, most of the fields for the Node
+message will be read from the bootstrap file.
+
+#### LDS
+
+The gRPC client will typically start by sending an LDS request for a
+`Listener` resource whose name matches the server name from the target
+URI used to create the gRPC channel (including port suffix if present).
+Note that the server name will also be used in the HTTP/2 ":authority"
+field, so it needs to be in host or host:port syntax.  However, the gRPC
+client will not interpret it in any special way.  This means that the
+xDS server effectively defines what the default port means if the URI
+does not specify a port.
+
+Because we are requesting one specific resource by name, the LDS response
+should include at least one `Listener`.  However, many existing xDS servers
+do not support requesting one specific resource in LDS, so the gRPC client
+must be tolerant of servers that may ignore the requested resource name;
+if the server returns multiple resources, the client will look for the one
+with the name it asked for, and it will ignore the rest of the entries.
+Note that this means that the resource returned by the xDS server must
+have exactly the name specified by the client.
+
+#### Listener Proto
+
+The returned `Listener` should be an "HTTP API listener", using the format
+added to the xDS API in https://github.com/envoyproxy/envoy/pull/8170.
+The HTTP API listener configuration may either provide the
+`RouteConfiguration` directly inline, or it may tell the client to
+use RDS.  Note that if using RDS, the __ConfigSource__ proto for the RDS
+server must indicate to use ADS.
+
+#### RDS
+
+If the `Listener` does not include the `RouteConfiguration` inline,
+then the client will send an RDS request asking for the specific
+`RouteConfiguration` name specified in the `Listener`.
+
+Because we are requesting one specific resource by name, the RDS response
+should include no more than one RouteConfiguration.  However, the gRPC
+client must be tolerant of servers that may ignore the resource name in
+the request; if the server returns multiple resources, the client will
+look for the one with the name it asked for, and it will ignore the rest
+of the entries.
+
+#### RouteConfiguration Proto
+
+The `RouteConfiguration` includes a list of zero or more __VirtualHost__
+resources.  The gRPC client will look through this list to find an element
+whose domains field matches the server name specified in the "xds:" URI
```

Comment:
Include how domains are matched?

Reply:
I don't think it's necessary to spell that out here, since we basically just do the same thing as Envoy.